### PR TITLE
Add sanity tests

### LINKED ISSUE #64: Remove requirement on urllib3 and requests
Since **urllib3** and **requests** is very slow to be imported on Raspberry Pi, it would be better to remove it from the source code and implement **urllib**/**urllib2** support. Especially since the code is pretty simple and doesn't need any special functionality from requests.

### PR DESCRIPTION
**Pull request by dagwieers**
This PR includes:
- Add sanity testing, including Travis
- Fix all pylint and PEP8 issues
- Add pip requirements.txt
- Add a Makefile for creating a working ZIP package

This relates to #63

### CODE REVIEW

**Comment by dagwieers** [comment]
It would be nice if Travis integration could be enabled for the sanity testing.

**Comment by dagwieers** [comment]
I am sorry for mixing 2 things up in a single PR, but I started with removing Requests with urlib/urllib2 for performance reasons (removing 800ms hit for loading urllib3 on RPi).

And only when I had that done, I looked at testing this code easily which required some changes. An so I added PEP8 and pylint to this as well.

**Comment by mediaminister** [comment]
I checked your changes and saw you broke `_install_inputstream` function. Because `xbmc.executebuiltin` doesn't return if it succeeded or failed we have to check afterwards if the inputstream addon exists. I added this fix to this pull request already.

I will do a couple of more tests today and report back.

**Comment by emilsvennesson** [comment]
Please split the changes into separate commits before merging as well.

File: lib/inputstreamhelper.py [hunk 1]
```diff
@@ -801,9 +801,13 @@ def _check_drm(self):
 
     def _install_inputstream(self):
         """Install inputstream addon."""
```
Comment (dagwieers) [inline lib/inputstreamhelper.py]: Thanks! This must have been a merge issue, as this was properly tested.

Travis would have caught this too 😉

**Comment by dagwieers** [comment]
@emilsvennesson How do you want them to be split?

File: lib/inputstreamhelper.py (hunk 1 above)
Comment (mediaminister) [reply lib/inputstreamhelper.py]: No, it was not a merge issue. I didn't see the error in the sanity tests, I did a manual test with a LibreELEC virtual machine. This problem occurs **after** `xbmc.exeutebuiltin` fails to install the inputstream add-on. Later on in the script inputstreamhelper is doing a version check on an unexisting add-on and that causes a crash.

Comment (dagwieers) [reply lib/inputstreamhelper.py]: On my cellphone it appeared as if the try-statement was missing in my code.
That's why I thought this was basically a syntax error in the first place.

**Comment by dagwieers** [comment]
@emilsvennesson I don't mind separating the Requests-change from the sanity checks/fixes. (As it should have been), but separating the different commits make no sense as these are gradual fixes during development. Each change does not represent a new entity, so using the reflog to turn them in commits is a waste of everyone's time.

The amend-commits during development (in most cases) are there for testing, to ensure a new commit hash is used in the ZIP file so I can actually test individual changes, without having a new commit for every small change or typo.

**Comment by emilsvennesson** [comment]
@dagwieers Yeah, it was the requests change I'd like to see as a separate commit. I'm with you on the rest.

**Comment by dagwieers** [comment]
I think I managed to split both. It was dirty :-(
And now it needs to be retested, twice.

**Comment by mediaminister** [comment]
Added a couple of fixes. Looks good to me now.

**Comment by dagwieers** [comment]
I am going to merge this as it is holding everything back at the moment.

PS We could set rules on this repository that e.g. requires PRs to be reviewed or CI turns green before they can be merged etc.

**Comment by basrieter** [comment]
Seems reasonable

**Comment by mediaminister** [comment]
I'm okay with setting rules, but I guess that's something the repo owner has to do.

**Comment by dagwieers** [comment]
@mediaminister Yes, same for Travis. Only the owner can do this :-/

**Pull request merged**